tests/suggestion-box.test.ts

```diff
@@ -0,0 +1,309 @@
+import { fixture, expect, html, triggerFocusFor } from "@open-wc/testing";
+import "../src/suggestion-box.js";
+
+describe("SuggestionBox", () => {
+    it("should render input box", async () => {
+        const el = await fixture(html`<aalam-sgn-box> </aalam-sgn-box>`);
+        const input = el.shadowRoot?.querySelector("input");
+        expect(input).to.exist;
+    });
+
+    it("should render input box with id 'sgn-input'", async () => {
+        const el = await fixture(html`<aalam-sgn-box>
+            
+        </aalam-sgn-box>`);
+        const input1 = el.shadowRoot?.getElementById("sgn-input");
+        expect(input1).to.exist;
+    });
+
+    it("should handle focus and display of #sgn-container", async () => {
+        let el = await fixture(html`<aalam-sgn-box></aalam-sgn-box>`);
+        let input = el.shadowRoot?.querySelector("input");
+        let dd = el.shadowRoot?.querySelector("#sgn-container");
+        expect(dd?.style.display).to.equal("none");
+        input?.focus();
+        await el.updateComplete;
+        expect(dd?.style.display).to.equal("block");
+
+        el = await fixture(
+            html`<aalam-sgn-box
+                ><div slot="sgn-empty">
+                    This is the empty block
+                </div></aalam-sgn-box
+            >`
+        );
+        input = el.shadowRoot?.querySelector("input");
+        dd = el.shadowRoot?.querySelector("#sgn-container");
+        expect(dd?.style.display).to.equal("none");
+        input?.focus();
+        await el.updateComplete;
+        expect(dd?.style.display).to.equal("block");
+    });
+
+    it("should render #sgn-container", async () => {
+        const el = await fixture(html`<aalam-sgn-box></aalam-sgn-box>`);
+        const container = el.shadowRoot?.getElementById("sgn-container");
+        expect(container).to.exist;
+    });
+
+    it("should render #private-item", async () => {
+        const el = await fixture(html`<aalam-sgn-box></aalam-sgn-box>`);
+        const privateitem = el.shadowRoot?.getElementById("private-item");
+        expect(privateitem).to.exist;
+    });
+
+    it("should update suggestions on input event", async () => {
+        const el = await fixture(
+            html`
+                <aalam-sgn-box>
+                    <div slot="sgn-item-template">{name}</div>
+                </aalam-sgn-box>
+            `
+        );
+        const input = el.shadowRoot?.querySelector("input");
+        el.list = [
+            {
+                name: "thiruvarur",
+                description: "My native",
+            },
+            {
+                name: "coimbatore",
+                description: "Working place",
+            },
+            {
+                name: "chennai",
+                description: "dream location",
+            },
+        ];
+        el.listkey = "name";
+        input.value = "thi";
+        input.dispatchEvent(
+            new Event("input", { bubbles: true, composed: true })
```

Comment:
I think instead of doing this way, you can send a keyboard command to the element. You can checkout how its used in the datepicker test cases. This is not reliable to test. 
You can also register listener on `el` for input and check how input event you received after typing